ledger-api/grpc-definitions/com/daml/ledger/api/v2/state_service.proto

```diff
@@ -0,0 +1,170 @@
+  // The format of this field is described in ``ledger_offset.proto``.
+  string offset = 1;
+
+  // The workflow ID used in command submission which corresponds to the contract_entry. Only set if
+  // the ``workflow_id`` for the command was set.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional
+  string workflow_id = 2;
+
+  // For a contract there could be multiple contract_entry-s in the entire snapshot. These together define
+  // the state of one contract in the snapshot.
+  // A contract_entry is only included in the result, iff there is at least one stakeholder party of the contract
+  // which is hosted on the synchronizing domain at the time of the underlying event and satisfies the
+  // ``TransactionFilter`` in the query.
+  oneof contract_entry {
+    // The contract is active on the given domain, meaning: there was an activation event on the given domain (
+    // created, assigned), which is not followed by a deactivation event (archived, unassigned) on the same
+    // domain - until the active_at_offset.
+    // Since activeness is defined as a per domain concept, it is possible, that a contract is active on one
+    // domain, but already archived on another.
+    // There will be one such message for each domain the contract is active on.
+    ActiveContract active_contract = 3;
+
+    // Included iff the unassigned event was before or at the active_at_offset, but there was no corresponding
+    // assigned event before or at the active_at_offset.
+    IncompleteUnassigned incomplete_unassigned = 4;
+
+    // Important: this message is not indicating that the contract is active on the target domain!
+    // Included iff the assigned event was before or at the active_at_offset, but there was no corresponding
+    // unassigned event before or at the active_at_offset.
+    IncompleteAssigned incomplete_assigned = 5;
+  }
+}
+
+message ActiveContract {
+
+  // Required
+  com.daml.ledger.api.v1.CreatedEvent created_event = 1;
+
+  // A valid domain ID
+  // Required
+  string domain_id = 2;
+
+  // Each corresponding assigned and unassigned event has the same reassignment_counter. This strictly increases
+  // with each unassign command for the same contract. Creation of the contract corresponds to reassignment_counter
+  // equals zero.
+  // This field will be the reassignment_counter of the latest observable activation event on this domain, which is
+  // before the active_at_offset.
+  // Required
+  uint64 reassignment_counter = 3;
+}
+
+message IncompleteUnassigned {
+
+  // Required
+  com.daml.ledger.api.v1.CreatedEvent created_event = 1;
+
+  // Required
+  UnassignedEvent unassigned_event = 2;
+}
+
+message IncompleteAssigned {
+
+  // Required
+  AssignedEvent assigned_event = 1;
+}
+
+
+message GetConnectedDomainsRequest {
+
+  // The submitting party
+  // Must be a valid PartyIdString (as described in ``value.proto``).
+  // Required
+  string party = 1;
```

Comment:
```suggestion
  // The party of interest
  // Must be a valid PartyIdString (as described in ``value.proto``).
  // Required
  string party = 1;
```
@cocreature : when will we need the `GetConnectedDomains` endpoint in our apps once we have merged streams? If we need it, then will we poll these for every party of interest?

Reply:
changed it thanks

Reply:
@meiersi-da we need them to resolve domain alias -> domain ids in our config.

Reply:
OK. I know I don't fully understand the ramifications of the choice to make this query single-party, but it seems like a sensible starting point. Resolving.